### PR TITLE
cp2k : added dependencies

### PR DESCRIPTION
**Pull request by alalazo**
##### Modifications
- [x] added `wannier90` and `elpa` as dependencies of `cp2k`

### CODE REVIEW

**Comment by alalazo** [comment]
@tgamblin If you could merge #1264 from @adamjstewart I also integrated that into `cp2k`

File: var/spack/repos/builtin/packages/cp2k/package.py [hunk 1]
```diff
@@ -143,15 +143,25 @@ def install(self, spec, prefix):
...
+                        'elpa-{0}'.format(str(spec['elpa'].version)),
+                        'modules'
+                    ),
                     '-I' + join_path(spec['pexsi'].prefix, 'fortran')
                 ])
                 ldflags.extend([
                     '-L' + spec['scalapack'].prefix.lib
                 ])
                 libs.extend([
+                    join_path(spec['elpa'].prefix.lib, 'libelpa.so'),
```
Comment (davydden) [inline var/spack/repos/builtin/packages/cp2k/package.py]: `.so` won't work on macOS.

File: var/spack/repos/builtin/packages/wannier90/package.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+    parallel = False
+
+    def install(self, spec, prefix):
+
+        substitutions = {
+            '@F90': spack_fc,
+            '@MPIF90': self.spec['mpi'].mpifc,
+            '@LIBS': ' '.join([
+                self.spec['lapack'].lapack_shared_lib,
```
Comment (davydden) [inline var/spack/repos/builtin/packages/wannier90/package.py]: maybe wait unitil your blas/lapack PR is merged to update this?

Comment (alalazo) [inline var/spack/repos/builtin/packages/wannier90/package.py]: I would, but my recent experience with #1186 is that changes that affect core may take a long time to be reviewed...

File: var/spack/repos/builtin/packages/cp2k/package.py (hunk 1 above)
Comment (alalazo) [inline var/spack/repos/builtin/packages/cp2k/package.py]: Definitely a place for `find_libraries` from #1682

File: var/spack/repos/builtin/packages/wannier90/package.py (hunk 2 above)
Comment (davydden) [inline var/spack/repos/builtin/packages/wannier90/package.py]: true...

File: var/spack/repos/builtin/packages/cp2k/package.py (hunk 1 above)
Comment (alalazo) [inline var/spack/repos/builtin/packages/cp2k/package.py]: Done along with other two `dso_suffix` that I overlooked... thanks

**Pull request merged**